TST: WCSSUB_TIME broke aarch64 and s390x jobs

Example log: https://github.com/astropy/astropy/runs/6059250593?check_suite_focus=true

```
    astropy/wcs/src/wcslib_wrap.c: In function ‘PyWcsprm_sub’:
    astropy/wcs/src/wcslib_wrap.c:1899:25: error: ‘WCSSUB_TIME’ undeclared (first use in this function); did you mean ‘WCSSUB_STOKES’?
     1899 |           element_val = WCSSUB_TIME;
          |                         ^~~~~~~~~~~
          |                         WCSSUB_STOKES
    astropy/wcs/src/wcslib_wrap.c:1899:25: note: each undeclared identifier is reported only once for each function it appears in
    astropy/wcs/src/wcslib_wrap.c: In function ‘_setup_wcsprm_type’:
    astropy/wcs/src/wcslib_wrap.c:4211:14: error: ‘WCSSUB_TIME’ undeclared (first use in this function); did you mean ‘WCSSUB_STOKES’?
     4211 |     CONSTANT(WCSSUB_TIME)      ||
          |              ^~~~~~~~~~~
    astropy/wcs/src/wcslib_wrap.c:4160:52: note: in definition of macro ‘CONSTANT’
     4160 | #define CONSTANT(a) PyModule_AddIntConstant(m, #a, a)
          |                                                    ^
    astropy/wcs/src/wcslib_wrap.c:4276:1: warning: control reaches end of non-void function [-Wreturn-type]
     4276 | }
          | ^
    error: command '/usr/bin/aarch64-linux-gnu-gcc' failed with exit code 1
```

Suspect:

* #13062

These jobs do not use the bundled WCSLIB library:

https://github.com/astropy/astropy/blob/637e0619d8dc8370843e9a05dad783f17e4a7cf4/.github/workflows/ci_cron_weekly.yml#L128-L129

--- a/astropy/wcs/docstrings.py
+++ b/astropy/wcs/docstrings.py
@@ -2312,7 +2312,7 @@ axes : int or a sequence.
 
     - ``'stokes'`` / ``WCSSUB_STOKES``: Stokes axis
 
-    - ``'temporal'`` / WCSSUB_TIME: Time axis
+    - ``'temporal'`` / ``WCSSUB_TIME``: Time axis (requires ``WCSLIB`` version 7.8 or greater)
 
     - ``'celestial'`` / ``WCSSUB_CELESTIAL``: An alias for the
       combination of ``'longitude'``, ``'latitude'`` and ``'cubeface'``.

--- a/astropy/wcs/src/wcslib_wrap.c
+++ b/astropy/wcs/src/wcslib_wrap.c
@@ -1833,24 +1833,27 @@ PyWcsprm_sub(
     PyObject* args,
     PyObject* kwds) {
 
-  int        i              = -1;
-  Py_ssize_t tmp            = 0;
-  PyObject*  py_axes        = NULL;
-  PyWcsprm*  py_dest_wcs    = NULL;
-  PyObject*  element        = NULL;
-  PyObject*  element_utf8   = NULL;
-  char*      element_str    = NULL;
-  int        element_val    = 0;
-  int        nsub           = 0;
-  int*       axes           = NULL;
-  int        status         = -1;
-  const char*    keywords[] = {"axes", NULL};
+  int        i            = -1;
+  Py_ssize_t tmp          = 0;
+  PyObject*  py_axes      = NULL;
+  PyWcsprm*  py_dest_wcs  = NULL;
+  PyObject*  element      = NULL;
+  PyObject*  element_utf8 = NULL;
+  char*      element_str  = NULL;
+  int        element_val  = 0;
+  int        nsub         = 0;
+  int*       axes         = NULL;
+  int        status       = -1;
+  int        wcslib_ver[] = {0, 0, 0};
+  const char* keywords[]  = {"axes", NULL};
 
   if (!PyArg_ParseTupleAndKeywords(
           args, kwds, "|O:sub", (char **)keywords,
           &py_axes)) {
     goto exit;
   }
+
+  wcslib_version(wcslib_ver);
 
   if (py_axes == NULL || py_axes == Py_None) {
     /* leave all variables as is */
@@ -1895,14 +1898,30 @@ PyWcsprm_sub(
           element_val = WCSSUB_SPECTRAL;
         } else if (strncmp(element_str, "stokes", 7) == 0) {
           element_val = WCSSUB_STOKES;
-        } else if (strncmp(element_str, "temporal", 9) == 0) {
+#if defined(WCSSUB_TIME)
+        } else if ((wcslib_ver[0] > 7 || (wcslib_ver[0] == 7 && wcslib_ver[1] >= 8)) &&
+                   strncmp(element_str, "temporal", 9) == 0) {
           element_val = WCSSUB_TIME;
+#else
+        } else if (strncmp(element_str, "temporal", 9) == 0) {
+          PyErr_Format(
+            PyExc_NotImplementedError,
+            "Support for 'temporal' axis requires WCSLIB version 7.8 or greater "\
+            "but linked WCSLIB version is %s", wcslib_version(NULL)
+          );
+          goto exit;
+#endif
         } else if (strncmp(element_str, "celestial", 10) == 0) {
           element_val = WCSSUB_CELESTIAL;
         } else {
           PyErr_SetString(
             PyExc_ValueError,
-            "string values for axis sequence must be one of 'latitude', 'longitude', 'cubeface', 'spectral', 'stokes', or 'celestial'");
+#if defined(WCSSUB_TIME)
+            "string values for axis sequence must be one of 'latitude', 'longitude', 'cubeface', 'spectral', 'stokes', 'temporal', or 'celestial'"
+#else
+            "string values for axis sequence must be one of 'latitude', 'longitude', 'cubeface', 'spectral', 'stokes', or 'celestial'"
+#endif
+            );
           goto exit;
         }
         Py_CLEAR(element_utf8);
@@ -4208,7 +4227,9 @@ _setup_wcsprm_type(
     CONSTANT(WCSSUB_CUBEFACE)  ||
     CONSTANT(WCSSUB_SPECTRAL)  ||
     CONSTANT(WCSSUB_STOKES)    ||
+#if defined(WCSSUB_TIME)
     CONSTANT(WCSSUB_TIME)      ||
+#endif
     CONSTANT(WCSSUB_CELESTIAL) ||
     CONSTANT(WCSHDR_IMGHEAD)   ||
     CONSTANT(WCSHDR_BIMGARR)   ||

--- a/astropy/wcs/tests/test_tab.py
+++ b/astropy/wcs/tests/test_tab.py
@@ -61,11 +61,17 @@ def test_mixed_celest_and_1d_tab_roundtrip():
     assert np.allclose(pts, w.wcs_world2pix(w.wcs_pix2world(pts, 0), 0))
 
 
+@pytest.mark.skipif(
+    _WCSLIB_VER < Version('7.8'),
+    reason="Requires WCSLIB >= 7.8 for swapping -TAB axes to work."
+)
 def test_wcstab_swapaxes():
     # Crash on deepcopy of swapped -TAB axes reported in #13036.
     # Fixed in #13063.
     filename = get_pkg_data_filename('data/tab-time-last-axis.fits')
     with fits.open(filename) as hdul:
         w = wcs.WCS(hdul[0].header, hdul)
+        w.wcs.ctype[-1] = 'FREQ-TAB'
+        w.wcs.set()
     wswp = w.swapaxes(2, 0)
     deepcopy(wswp)

--- a/astropy/wcs/wcs.py
+++ b/astropy/wcs/wcs.py
@@ -38,7 +38,8 @@ import textwrap
 import warnings
 import builtins
 
-# THIRD-PARTY
+# THIRD-
+from packaging.version import Version
 import numpy as np
 
 # LOCAL
@@ -70,8 +71,7 @@ __doctest_skip__ = ['WCS.all_world2pix']
 
 
 if _wcs is not None:
-    _parsed_version = _wcs.__version__.split('.')
-    if int(_parsed_version[0]) == 5 and int(_parsed_version[1]) < 8:
+    if Version(_wcs.__version__) < Version("5.8"):
         raise ImportError(
             "astropy.wcs is built with wcslib {0}, but only versions 5.8 and "
             "later on the 5.x series are known to work.  The version of wcslib "
@@ -81,6 +81,10 @@ if _wcs is not None:
         raise RuntimeError(
             "astropy.wcs did not pass its sanity check for your build "
             "on your platform.")
+
+    _WCSSUB_TIME_SUPPORT = Version(_wcs.__version__) >= Version("7.8")
+    _WCS_TPD_WARN_LT71 = Version(_wcs.__version__) < Version("7.1")
+    _WCS_TPD_WARN_LT74 = Version(_wcs.__version__) < Version("7.4")
 
     WCSBase = _wcs._Wcs
     DistortionLookupTable = _wcs.DistortionLookupTable
@@ -140,6 +144,10 @@ else:
     NonseparableSubimageCoordinateSystemError = None
     NoWcsKeywordsFoundError = None
     InvalidTabularParametersError = None
+
+    _WCSSUB_TIME_SUPPORT = False
+    _WCS_TPD_WARN_LT71 = False
+    _WCS_TPD_WARN_LT74 = False
 
 
 # Additional relax bit flags
@@ -2684,7 +2692,7 @@ reduce these to 2 dimensions using the naxis kwarg.
                 if kw in header:
                     del header[kw]
             # Check if we can handle TPD distortion correctly
-            if int(_parsed_version[0]) * 10 + int(_parsed_version[1]) < 71:
+            if _WCS_TPD_WARN_LT71:
                 for kw, val in header.items():
                     if kw[:5] in ('CPDIS', 'CQDIS') and val == 'TPD':
                         warnings.warn(
@@ -2692,7 +2700,7 @@ reduce these to 2 dimensions using the naxis kwarg.
                             f"{_wcs.__version__} is writing this in a format incompatible with "
                             f"current versions - please update to 7.4 or use the bundled WCSLIB.",
                             AstropyWarning)
-            elif int(_parsed_version[0]) * 10 + int(_parsed_version[1]) < 74:
+            elif _WCS_TPD_WARN_LT74:
                 for kw, val in header.items():
                     if kw[:5] in ('CPDIS', 'CQDIS') and val == 'TPD':
                         warnings.warn(
@@ -3231,6 +3239,12 @@ reduce these to 2 dimensions using the naxis kwarg.
         """
         A copy of the current WCS with only the time axes included
         """
+        if not _WCSSUB_TIME_SUPPORT:
+            raise NotImplementedError(
+                "Support for 'temporal' axis requires WCSLIB version 7.8 or "
+                f"greater but linked WCSLIB version is {_wcs.__version__}"
+            )
+
         return self.sub([WCSSUB_TIME])  # Defined by C-ext  # noqa: F821
 
     @property

